frontend2/src/components/elements/Icon.tsx

```diff
@@ -0,0 +1,85 @@
+import React from "react";
+import {
+  ClipboardDocumentIcon as ClipboardDocumentIcon24,
+  HomeIcon as HomeIcon24,
+  MapIcon as MapIcon24,
+  TrophyIcon as TrophyIcon24,
+  ChartBarIcon as ChartBarIcon24,
+  ClockIcon as ClockIcon24,
+  UserGroupIcon as UserGroupIcon24,
+  ArrowUpTrayIcon as ArrowUpTrayIcon24,
+  PlayCircleIcon as PlayCircleIcon24,
+  ChevronDownIcon as ChevronDownIcon24,
+  CheckIcon as CheckIcon24,
+  InformationCircleIcon as InformationCircleIcon24,
+} from "@heroicons/react/24/outline";
+
+import {
+  ClipboardDocumentIcon as ClipboardDocumentIcon20,
+  HomeIcon as HomeIcon20,
+  MapIcon as MapIcon20,
+  TrophyIcon as TrophyIcon20,
+  ChartBarIcon as ChartBarIcon20,
+  ClockIcon as ClockIcon20,
+  UserGroupIcon as UserGroupIcon20,
+  ArrowUpTrayIcon as ArrowUpTrayIcon20,
+  PlayCircleIcon as PlayCircleIcon20,
+  ChevronDownIcon as ChevronDownIcon20,
+  CheckIcon as CheckIcon20,
+  InformationCircleIcon as InformationCircleIcon20,
+} from "@heroicons/react/20/solid";
+
+const icons24 = {
+  clipboard_document: ClipboardDocumentIcon24,
+  home: HomeIcon24,
+  map: MapIcon24,
+  trophy: TrophyIcon24,
+  chart_bar: ChartBarIcon24,
+  clock: ClockIcon24,
+  user_group: UserGroupIcon24,
+  arrow_up_tray: ArrowUpTrayIcon24,
+  play_circle: PlayCircleIcon24,
+  chevron_down: ChevronDownIcon24,
+  check: CheckIcon24,
+  information_circle: InformationCircleIcon24,
+};
+
+const icons20 = {
```

Comment:
we need both 24 and 20? would it be worth it to cut down and only need to import one of them?

Reply:
(jumping in here as someone who hasn't been paying attention and who has just read some email digests)

I think it's ok to have both, because it makes sense to be able to tailor/change icon-sets for different episodes of Battlecode. But, this is conditioned on:

- The setup being something easily extensible to *n* episodes of Battlecode for large *n*.
- I think it's ok if this extensibility means defining a new dict/mapping for each episode, because that's unavoidable when each episode could be different.
- However, it should also be implementable without changing the downstream rendering code, so [this](https://github.com/battlecode/galaxy/pull/670/files/444aa6f45c9ef9425827eb8ae2217fc4690f8504#diff-b9386fd95d3bd66ff698696964d7a7e70d9acdf0199982091a61340678d36736R62) looks a bit more dubious. Perhaps it could be a single mapping that maps each episode name to one of these dicts?

Separately, curious how easy it would be to use this framework to add new menu items in the future with new icons? (possibly the answer is "very easy", I didn't read the code at all)

Reply:
(oops, 24/20 px not episode?)

Reply:
ah, if either size gets used for different purposes, then sounds good
I'm a little wary of the extra overhead here -- if we want to add a new icon, then we'd have to import both sizes, and make sure the correct size gets used. But perhaps this is easier than I think. Most of all though, we could always come back to this later

Reply:
ah fair, naming wasn't entirely clear but makes sense now